ydb/core/kqp/compile_service/kqp_compile_service.cpp

```diff
@@ -51,8 +51,10 @@ class TKqpQueryCache {
         AstIndex.emplace(GetQueryIdWithAst(*compileResult->Query, *compileResult->Ast), compileResult->Uid);
     }
 
-    bool Insert(const TKqpCompileResult::TConstPtr& compileResult, bool isEnableAstCache) {
-        InsertQuery(compileResult);
+    bool Insert(const TKqpCompileResult::TConstPtr& compileResult, bool isEnableAstCache, bool isPerStatement) {
+        if (!isPerStatement) {
```

Comment:
We won't save compilation result for statements by text, because all statements have the same query text, we will save their compilation results by ast